### ISSUE TITLE
Question: a good strategy to estimate covariance matrix for SNGP when there are no clear "epochs"

### ISSUE THREAD

**Issue by hr0nix** [google/uncertainty-baselines#1098]
Hello!

What strategy would you suggest for estimating the covariance matrix for SNGP in a setting where we have a continuous flow of training data (for example, in reinforcement learning)? I guess I can try different values of momentum, but it's a bit tricky to pick the right value even in simplest settings.

Thanks!

**Comment by ramanujan1729** [comment]
Hi hr0nix, 

See the footnote 5 on page 11 of the [extended version of the paper](https://arxiv.org/pdf/2205.00403). You have it [here in edward2](https://github.com/google/edward2/blob/main/edward2/tensorflow/layers/random_feature.py#L394). 
Basically the idea is to update it by batch, using momentum parameter - similarly to the way they do it in [Batch Norm paper](https://arxiv.org/pdf/1502.03167).

I hope this helps!
